staging/src/k8s.io/apimachinery/pkg/util/yaml/decoder.go

```diff
@@ -98,7 +99,7 @@ func (d *YAMLToJSONDecoder) Decode(into interface{}) error {
 	}
 
 	if len(bytes) != 0 {
-		err := yaml.Unmarshal(bytes, into)
+		err := yaml3.Unmarshal(bytes, into)
```

Comment:
`sigs.k8s.io/yaml` uses json tags in structs for marshaling/unmarshaling, and does mechanical yaml-json conversion

`gopkg.in/yaml.v3` expects `yaml` struct tags, which none of our API types have, so this switch will not work at all for API types.